app/forms/application_form.rb

```diff
@@ -12,4 +11,22 @@ def self.model_name
     # This allows Rails magic such as route paths.
     ActiveModel::Name.new(self, nil, to_s.delete_suffix(FORM_CLASS_SUFFIX))
   end
+
+  def self.accepts_nested_attributes_for(*models, **_options)
+    models.each do |model|
+      attr_accessor(model)
+
+      model_ivar_symbol = :"@#{model}"
+      model_class = model.to_s.classify.constantize
+
+      define_method(:"#{model}_attributes=") do |attributes|
+        instance_variable_set(model_ivar_symbol, instance_variable_get(model_ivar_symbol) || [])
+
+        pp(model_ivar_symbol)
+        pp(model_class)
+        pp(attributes)
```

Comment:
Left debugging code in for now since I'll have more work to do if we decide to pursue this approach.